src/Components/Components/src/Routing/LocationChangingContext.cs

```diff
@@ -8,17 +8,16 @@ namespace Microsoft.AspNetCore.Components.Routing;
 /// </summary>
 public class LocationChangingContext
 {
-    private readonly CancellationTokenSource _cts;
-
-    internal LocationChangingContext(string targetLocation, string? historyEntryState, bool isNavigationIntercepted, CancellationTokenSource cts)
+    internal LocationChangingContext(string targetLocation, string? historyEntryState, bool isNavigationIntercepted, CancellationToken cancellationToken)
```

Comment:
One thing to consider here, is how would people write unit tests for this. Does this type need to be "constructible" by people in unit tests?

Reply:
I believe this discussion was had [here](https://github.com/dotnet/aspnetcore/pull/42638#discussion_r923501645). The conclusion seems to be that it's fine if this stays `internal`, at least for the bUnit case.